bpf/bpf_wireguard.c

```diff
@@ -95,33 +108,20 @@ handle_ipv6(struct __ctx_buff *ctx, __u32 identity, __s8 *ext_err __maybe_unused
 	}
 #endif
 
-#ifndef ENABLE_HOST_ROUTING
-	return CTX_ACT_OK;
-#endif
```

Comment:
I believe from a netfilter-symmetry perspective, we need to step carefully here.

In `bpf_overlay` it's easy - we expect that both legs of the connection flow via the overlay network. And so packets by the local pod will get redirect to `cilium_vxlan` immediately, with no netfilter traversal. And therefore we can also unconditionally bypass netfilter for packets coming out of the overlay, towards the local pod.

But with Wireguard we do have the case (when BPF Host Routing is disabled) where a local pod's traffic first passes through netfilter, and only in `to-netdev` gets redirected to `cilium_wg`. And so the `from-wireguard` path also needs to traverse netfilter, and can't unconditionally redirect straight to the pod.